test_runner/fixtures/utils.py

```diff
@@ -89,3 +90,40 @@ def get_dir_size(path: str) -> int:
                 pass  # file could be concurrently removed
 
     return totalbytes
+
+
+def get_timeline_dir_size(path: pathlib.Path) -> int:
+    """Get the timeline directory's total size, which only counts the layer files' size."""
+    sz = 0
+    for f_name in os.listdir(path):
```

Comment:
nit: Path has handy `iterdir` method that yields directly Path instances so there is no need to construct full path with directory manually.

So this is the full snippet:
```python
    for dir_entry in path.iterdir():
        with contextlib.suppress(Exception):
            # file is an image layer
            _ = parse_image_layer(dir_entry.name)
            sz += dir_entry.stat().st_size
            continue
        
        with contextlib.suppress(Exception):
            # file is a delta layer
            _ = parse_delta_layer(dir_entry.name)
            sz += dir_entry.stat().st_size
            continue
```
To me it looks a bit cleaner with suppress but feel free to keep try except